### DISCUSSION TITLE
Kazakh Character Encoding Request for OpenUTAU

### DISCUSSION THREAD

**Discussion by biscript** [stakira/OpenUtau#1065] [Ideas]
To the esteemed development team of OpenUTAU,

I humbly request the addition of Kazakh character encoding support to OpenUTAU. While Kazakh employs Cyrillic script, it possesses a unique phonetic system that can be optimally utilized in vocal synthesis.

Kazakh features several distinctive phonemes, such as hard and soft consonants, which contribute to the richness and emotional depth of its speech and music. Supporting these phonemes in OpenUTAU would facilitate more natural and high-quality reproduction of Kazakh vocals and songs.

Furthermore, extending OpenUTAU's capabilities to include Kazakh character encoding would mark a significant step in supporting the music community in Kazakhstan and Central Asia. This support would enable musicians and producers from this region to better express their ideas and creative concepts.

I believe that expanding Open UTAU's features to encompass Kazakh character encoding would be beneficial to many musicians and producers, making our music community more diverse and inclusive.

I sincerely hope for your understanding and support regarding my request.

Best regards, DJaeB

**Comment by oxygen-dioxide** [comment]
For character encoding, since OpenUtau use UTF-8 encoding, you can already input Cyrillic characters into OpenUtau. There are Russian phonemizers that accept Cyrillic script.
To add native support for a new language, we need to:
- write a reclist and oto template for the language. It should contain all possible combinations of phonemes in the language.
- develop a phonemizer for the language that auto converts natural language words into phonemes.

You can check out existing examples, such as English VCCV and English X-sampa.

**Reply by biscript** [reply]
Yes, I understand that it would be possible to use UTF-8 encoding or Russian encoding, but the Kazakh language has nine additional characters that are not perceived by the above encodings. Namely: ә, і, ң, ғ, ү, ұ, қ, ө, һ ( do not confuse with English h ).

As for the oto template, it exists. I have my own type for banks KV1 ( Kazakh Vocal ver1 ), created taking into account the peculiarities of the Kazakh language.

Please also take a look at the links shown below, they contain all the information about Kazakh sound. 

## Resource:

1. All about the Kazakh sound: https://drive.google.com/drive/folders/1amn530gpAhGJg76I9hFnPabX5eYmc_VO
